src/services/resource.js

```diff
@@ -210,6 +358,364 @@ module.exports = class resourceHelper {
+				)
+			}
+
+			const resourceFilter = {
+				id: { [Op.in]: finalResourceIds },
+				organization_id: { [Op.in]: uniqueOrganizationIds },
+			}
+			if (searchText != '')
+				resourceFilter.title = {
+					[Op.iLike]: '%' + searchText + '%',
+				}
+
+			if (common.TYPE in queryParams && queryParams[common.TYPE]) {
+				resourceFilter.type = queryParams[common.TYPE].split(',')
+			}
+
+			const response = await resourceQueries.resourceList(
+				resourceFilter,
+				[
+					'id',
+					'title',
+					'type',
+					'organization_id',
+					'status',
+					'user_id',
+					'submitted_on',
+					'last_reviewed_on',
+					'created_at',
+				],
+				sort,
+				page,
+				limit
+			)
+
+			const uniqueCreatorIds = utils.getUniqueElements(
+				response.result.map((item) => {
+					return item.user_id
+				})
+			)
+
+			const userDetails = await this.fetchUserDetails(uniqueCreatorIds)
+			const orgDetails = await this.fetchOrganizationDetails(uniqueOrganizationIds)
+
+			resourcesIdsNotes = await this.fetchReviewerNotesForResources(
+				user_id,
+				finalResourceIds,
+				uniqueOrganizationIds
+			)
+
+			result.data = response.result.map((item) => {
+				let returnValue = item
+				returnValue.notes = resourcesIdsNotes[item.id] || ''
+				if (inProgressResources.includes(item.id)) {
+					returnValue.reviewer_status = common.REVIEW_STATUS_INPROGRESS
+				}
+				returnValue.creator =
+					userDetails[item.user_id] && userDetails[item.user_id].name ? userDetails[item.user_id].name : ''
+				returnValue.organization = orgDetails[item.organization_id]
+				delete item.user_id
+				delete item.organization_id
+				return returnValue
+			})
+			result.count = response.count
+			result.in_progress_count = in_progress_count
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'RESOURCE_LISTED_SUCCESSFULLY',
+				result,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findSequentialResources(organization_id, roles, resourceTypesInSequentialReview = []) {
+		// get unique user roles
+		const userRoleTitles = utils.getUniqueElements(roles.map((item) => item.title))
+
+		// fetch the resource wise review levels
+		const resourceWiseLevels = await this.fetchResourceReviewLevel(
+			organization_id,
+			userRoleTitles,
+			resourceTypesInSequentialReview
+		)
+		let innerArr = []
+		resourceTypesInSequentialReview.filter((item) => {
+			if (resourceWiseLevels[item]) {
+				innerArr.push({
+					[Op.and]: [{ type: item }, { next_stage: resourceWiseLevels[item] }],
+				})
+			}
+		})
+		const resourceFilter = {
+			[Op.or]: innerArr,
+			organization_id,
+			review_type: common.REVIEW_TYPE_SEQUENTIAL,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.resourceList(resourceFilter, ['id'])
+		const resoureId = resourcesDetails.result.map((item) => {
+			return item.id
+		})
+		return resoureId
```

Comment:
@adithyadinesh0412 return empty array and if resourcesDetails is empty then you dont need to map it